tests/posix/env/testcase.yaml

```diff
@@ -0,0 +1,31 @@
+common:
+  filter: not (CONFIG_NATIVE_BUILD and CONFIG_EXTERNAL_LIBC)
+  integration_platforms:
+    - qemu_riscv64
+  tags: posix
+tests:
+  portability.posix.env: {}
+  portability.posix.env.armclang_std_libc:
+    toolchain_allow: armclang
+    extra_configs:
+      - CONFIG_ARMCLANG_STD_LIBC=y
+  portability.posix.env.arcmwdtlib:
+    toolchain_allow: arcmwdt
+    extra_configs:
+      - CONFIG_ARCMWDT_LIBC=y
+  portability.posix.env.minimal:
+    extra_configs:
+      - CONFIG_MINIMAL_LIBC=y
+      - CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=256
+  portability.posix.env.newlib:
+    filter: TOOLCHAIN_HAS_NEWLIB == 1
+    extra_configs:
+      - CONFIG_NEWLIB_LIBC=y
+  portability.posix.env.picolibc:
+    tags: picolibc
+    platform_exclude:
+      - native_sim
+      - native_sim_64
```

Comment:
These are excluded, because it was the 1 test case that was not passing.

There is an error about `_POSIX_C_SOURCE` being redefined, likely here
https://github.com/zephyrproject-rtos/picolibc/blob/1a5c603b9f8e228f9459bdafedad15ea28efc700/zephyr/zephyr.cmake#L95

```shell
[927/1025] Building C object modules/picolibc/CMakeFiles/c.dir/newlib/libm/complex/clogl.c.obj
[928/1025] Building C object modules/picolibc/CMakeFiles/c.dir/newlib/libm/complex/cpowl.c.obj
[929/1025] Building C object modules/picolibc/CMakeFiles/c.dir/newlib/libm/complex/csinhl.c.obj
[930/1025] Building C object modules/picolibc/CMakeFiles/c.dir/newlib/libm/complex/csinl.c.obj
[931/1025] Building C object modules/picolibc/CMakeFiles/c.dir/newlib/libm/complex/ctanl.c.obj
[932/1025] Building C object modules/picolibc/CMakeFiles/c.dir/newlib/libm/complex/ctanhl.c.obj
[933/1025] Linking C static library modules/picolibc/libc.a
[934/1025] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[935/1025] Building C object zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj
FAILED: zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj 
ccache /usr/lib/ccache/gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DTC_RUNID=6b8855be30e94afb6f819ebb6f871902 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/home/cfriedt/zephyrproject/zephyr/kernel/include -I/home/cfriedt/zephyrproject/zephyr/arch/posix/include -I/home/cfriedt/zephyrproject/zephyr/include -I/home/cfriedt/zephyrproject/zephyr/twister-out/native_sim/tests/posix/env/portability.posix.env.picolibc/zephyr/include/generated -I/home/cfriedt/zephyrproject/zephyr/soc/posix/inf_clock -I/home/cfriedt/zephyrproject/zephyr/boards/posix/native_sim -I/home/cfriedt/zephyrproject/zephyr/scripts/native_simulator/common/src/include -I/home/cfriedt/zephyrproject/zephyr/scripts/native_simulator/native/src/include -I/home/cfriedt/zephyrproject/zephyr/subsys/testsuite/include -I/home/cfriedt/zephyrproject/zephyr/subsys/testsuite/ztest/include -isystem /home/cfriedt/zephyrproject/zephyr/lib/libc/common/include -isystem /home/cfriedt/zephyrproject/zephyr/twister-out/native_sim/tests/posix/env/portability.posix.env.picolibc/modules/picolibc/picolibc/include -fno-strict-aliasing -Werror -Os -imacros /home/cfriedt/zephyrproject/zephyr/twister-out/native_sim/tests/posix/env/portability.posix.env.picolibc/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/cfriedt/zephyrproject/zephyr/tests/posix/env=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/cfriedt/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/cfriedt/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -m32 -msse2 -mfpmath=sse -fvisibility=hidden -nostdinc -isystem /usr/lib/gcc/x86_64-linux-gnu/11/include -ffreestanding -fno-builtin -D_POSIX_C_SOURCE=200809 -std=c11 -MD -MT zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/posix/core/offsets/offsets.c.obj -c /home/cfriedt/zephyrproject/zephyr/arch/posix/core/offsets/offsets.c
<command-line>: error: "_POSIX_C_SOURCE" redefined [-Werror]
<command-line>: note: this is the location of the previous definition
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

Reply:
This is due to building PICOLIBC from source, you can trigger it by building from source for any target:
```
cmake -DBOARD=qemu_x86 ../tests/posix/env/ -GNinja -DCONFIG_PICOLIBC=y -DCONFIG_PICOLIBC_USE_MODULE=y
```
So I would not blacklist native_sim, but address the issue instead.

Despite of that warning, the test executes and passes fine.

Reply:
> So I would not blacklist native_sim, but address the issue instead.

Already underway. https://github.com/picolibc/picolibc/issues/649

> Despite of that warning, the test executes and passes fine.

Of course. The warning is escalated to an error though in CI, which is problematic.

Of course as well, it should really only be set when posix functionality is activated via Kconfig which is maybe more problematic.